.github/workflows/nolints.yml

```diff
@@ -0,0 +1,102 @@
+          ./elan-init -y --default-toolchain none
+          echo "$HOME/.elan/bin" >> $GITHUB_PATH
+
+      - uses: actions/checkout@v3
+
+      # We update `Mathlib.lean` as a convenience here,
+      # but verify that this didn't change anything in the `check_imported` job.
+      - name: update Mathlib.lean
+        run: |
+          find Mathlib -name "*.lean" | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
+
+      - name: If using a lean-pr-release toolchain, uninstall
+        run: |
+          if [[ `cat lean-toolchain` =~ ^leanprover/lean4-pr-releases:pr-release-[0-9]+$ ]]; then
+            echo "Uninstalling transient toolchain `cat lean-toolchain`"
+            elan toolchain uninstall `cat lean-toolchain`
+          fi
+
+      - name: print lean and lake versions
+        run: |
+          lean --version
+          lake --version
+
+      - name: build cache
+        run: |
+          lake build cache
+
+      - name: prune ProofWidgets .lake
+        run: |
+          # The ProofWidgets release contains not just the `.js` (which we need in order to build)
+          # but also `.oleans`, which may have been built with the wrong toolchain.
+          # This removes them.
+          # See discussion at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/nightly-testing/near/411225235
+          rm -rf .lake/packages/proofwidgets/.lake/build/lib
+          rm -rf .lake/packages/proofwidgets/.lake/build/ir
+
+      - name: get cache
+        run: |
+          lake exe cache clean
+          lake exe cache get
+
+      - name: build mathlib
+        id: build
+        uses: liskin/gh-problem-matcher-wrap@v2
+        with:
+          linters: gcc
+          run: |
+            bash -o pipefail -c "env LEAN_ABORT_ON_PANIC=1 lake build -KCI | tee stdout.log"
+
+      - name: lint mathlib
+        id: lint
+        uses: liskin/gh-problem-matcher-wrap@v2
+        with:
+          linters: gcc
+          run: |
+            env LEAN_ABORT_ON_PANIC=1 lake exe runLinter Mathlib
+            mv nolints.json scripts/nolints.json
+            ./scripts/update-style-exceptions.sh
+            git diff
+
+      - name: configure git setup
+        run: |
+          git remote add origin-bot "https://leanprover-community-bot:${{ secrets.LCB_TOKEN }}@github.com/leanprover-community/mathlib.git"
+          git config user.email "leanprover.community@gmail.com"
+          git config user.name "leanprover-community-bot"
+
+          # By default, github actions overrides the credentials used to access any
+          # github url so that it uses the github-actions[bot] user.  We want to access
+          # github using a different username.
+          git config --unset http.https://github.com/.extraheader
+
+      - name: update nolints.json and style-exceptions.txt
+        run: ./scripts/update_nolints.sh
+        env:
+          DEPLOY_GITHUB_TOKEN: ${{ secrets.LCB_TOKEN }}
```

Comment:
What is LCB_TOKEN? I'm guessing it is new, since it isn't otherwise mentioned in the repo. Could it have a more descriptive name?

Reply:
As I wrote, I cargo-culted this from mathlib(3) --- so presumably, this is not needed.
Can you advise on the proper way to do this instead? I really don't know the CI setup well.

Reply:
@eric-wieser do you know the answer here?

Reply:
It stands for "Lean Community Bot"